Tools/autotest/quadplane.py

```diff
@@ -2164,6 +2164,90 @@ def WindEstimateConsistency(self):
                 self.progress("Wind estimates correlated")
                 break
 
+    def FullRateClimbout(self):
+        '''test the full rate climbout after VTOL Takeoff functions'''
+        # Set from examining the autotest with enable/disable flag.
+        ALTITUDE_CUTOFF_M = 35
+        CRUISE_ALTITUDE_CUTOFF = 120
+        WAIT_AFTER_CRUISE_DELAY_S = 10
+
+        # Mission planning items.
+        CRUISE_WP_NUM = 2
+
+        self.progress("Testing full rate climbout.")
+
+        def run_test(flag):
+            filename = "mission.txt"
+            self.progress("Testing %s with flag %s" % (filename, bool(flag)))
+            self.progress("Setting parameters.")
+
+            self.set_parameter("Q_FR_CLMBOUT", int(flag))
+
+            self.reboot_sitl()
+
+            # Load and fly mission.
+            self.progress(f"Flying mission {filename}")
+            self.load_mission(filename)
+            self.wait_ready_to_arm()
+
+            # Get the starting altitude to convert to agl.
+            starting_alt_m = self.mav.messages['VFR_HUD'].alt
+
+            self.arm_vehicle()
+            self.change_mode('AUTO')
+
+            self.progress("Waiting for Transition done")
```

Comment:
Technically this is racy, a context should be used to collect statustexts.

In this case it's not really a problem - it takes a *long* time to get to the transition, so a user would need an astonishingly fast computer to actually hit the problem.